### PR TITLE
feat: Show evaluated value in embeddedDatasource url

### LINKED ISSUE #821: [Bug] Evaluated value doesn't show in URL & Action Selector Fields
Solutions: 

Since the url field is a combo of datasource.url + path + query params we need a way to pass the data tree and use evaluated value instead of data tree path
(this will cause an extra eval as it will come from redux)

- Pass data tree inside url component
- Pass data tree inside api pane so it can be used by other fields too
- Update data tree path logic to combine these in the code editor (will not cause extra eval)

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

PR to show evaluated value in datasource URL. 

Fixes #821

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Cypress

- Jest

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/show-evaluated-value-in-action-url 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.2 **(-0.02)** | 37.07 **(-0.02)** | 34.06 **(-0.01)** | 55.72 **(-0.02)**
 :red_circle: | app/client/src/components/editorComponents/form/fields/EmbeddedDatasourcePathField.tsx | 29.63 **(-1.58)** | 2.06 **(-0.41)** | 3.57 **(-0.43)** | 31.97 **(-1.11)**
 :red_circle: | app/client/src/entities/DataTree/dataTreeAction.ts | 21.05 **(-3.95)** | 0 **(0)** | 0 **(0)** | 21.43 **(-5.84)**
 :red_circle: | app/client/src/utils/DynamicBindingUtils.ts | 83.52 **(-2.28)** | 69.64 **(-0.84)** | 75.86 **(-2.71)** | 82.82 **(-2.53)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/client/src/entities/DataTree/dataTreeAction.ts [hunk 1]
```diff
@@ -5,23 +5,45 @@ import {
...
     action.config.dynamicBindingPathList.length
   ) {
     dynamicBindingPathList = action.config.dynamicBindingPathList.map((d) => ({
       ...d,
-      key: `config.${d.key}`,
+      key: d.key === "datasourceUrl" ? d.key : `config.${d.key}`,
     }));
   }
+
+  // update datasource url for embedded datasource action
```
Comment (hetunandu) [inline app/client/src/entities/DataTree/dataTreeAction.ts:29]: This is not the right place to do this

**Review by hetunandu — CHANGES_REQUESTED** [review]
Don't create a new path to create this. Instead just read all the evaluated values in the EmbeddedDatasourceField and pass that to the evaluated value pane

File: app/client/src/entities/DataTree/dataTreeAction.ts (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/entities/DataTree/dataTreeAction.ts:29]: Could you kindly suggest where this should happen?

Comment (hetunandu) [reply app/client/src/entities/DataTree/dataTreeAction.ts:29]: `EmbeddedDatasourceField`

**Review by hetunandu — APPROVED** [review]

**Pull request merged**